tests/unit/moduleapi/rdbloadsave.tcl

```diff
@@ -0,0 +1,151 @@
+set testmodule [file normalize tests/modules/rdbloadsave.so]
+
+start_server {tags {"modules"}} {
+    proc format_command {args} {
+        set cmd "*[llength $args]\r\n"
+        foreach a $args {
+            append cmd "$[string length $a]\r\n$a\r\n"
+        }
+        set _ $cmd
+    }
+
+    r module load $testmodule
+
+    test "Module rdbloadsave sanity" {
+        r test.sanity
+
+        assert_error {*No such file*} {r test.rdbload sanity.rdb}
+
+        r set x 1
+        assert_equal OK [r test.rdbsave sanity.rdb]
+
+        r flushdb
+        assert_equal OK [r test.rdbload sanity.rdb]
+        assert_equal 1 [r get x]
+    }
+
+    test "Module rdbloadsave test with pipelining" {
+        r config set save ""
+        r config set loading-process-events-interval-bytes 1024
+        r flushdb
+
+        populate 50000 a 128
+        r set x 111
+        assert_equal [r dbsize] 50001
+
+        assert_equal OK [r test.rdbsave blabla.rdb]
+        r flushdb
+        assert_equal [r dbsize] 0
+
+        # Send commands with pipeline. First command will call RM_RdbLoad() in
+        # the command callback. While loading RDB, Redis can go to networking to
+        # reply -LOADING. By sending commands in pipeline, we verify it doesn't
+        # cause a problem.
+        # e.g. Redis won't try to process next message of the current client
+        # while it is in the command callback.
+        r write [format_command test.rdbload blabla.rdb]
+        r flush
+        r write [format_command get x]
+        r write [format_command dbsize]
+        r flush
```

Comment:
if you really want them to be processed in the same event loop better write them with a single `write` call, or even better, make sure to SIGSTOP and SIGCONT before and after writing.

Reply:
I just want it to try to process next message while it is inside `rdbLoad()`. 

I tried SIGSTOP/SIGCONT, looks like it won't trigger the issue I saw (Currently, with the above test, it triggers an assert without `protectClient()` inside `RM_RdbLoad()`). 

Probably, the reason is if it reads all messages to the client buffer, then, when `RM_RdbLoad()` goes back to the event loop, there is no new event for the client. So, it won't trigger the assert.